plottingcheck.py

```diff
@@ -0,0 +1,53 @@
+import numpy as np
+import matplotlib
+import matplotlib.pyplot as plt
+from matplotlib.pyplot import cm 
+import pandautils as pup       
+
+def _plot_X(train, test, y_train, y_test, w_train, w_test, varlist, feature):
+	'''Args:
+		train:
+		test:
+		feature: a string like 'Jet', 'Muon', 'Photon'
+	'''
+
+	w_train_ext = np.array(pup.flatten([[w]*(len(train[i,0])) for i, w in enumerate(w_train)]))
+	w_test_ext = np.array(pup.flatten([[w]*(len(test[i,0])) for i, w in enumerate(w_test)]))
+	y_train_ext = np.array(pup.flatten([[y]*(len(train[i,0])) for i, y in enumerate(y_train)]))
+	y_test_ext = np.array(pup.flatten([[y]*(len(test[i,0])) for i, y in enumerate(y_test)]))
+
+	column_counter=0
+	for key in varlist:
+		if key.startswith(feature):
+			flat_train = pup.flatten(train[:,column_counter])
+			flat_test = pup.flatten(test[:,column_counter])
+			matplotlib.rcParams.update({'font.size': 16})
+			fig = plt.figure(figsize=(11.69, 8.27), dpi=100)
+			bins = np.linspace(min(min(pup.flatten(train[:,column_counter])), 
+				min(pup.flatten(test[:,column_counter]))), 
+				max(max(pup.flatten(train[:,column_counter])), 
+				max(pup.flatten(test[:,column_counter]))), 30)

```

Comment:
you already calculated `flatten(train[:, column_counter])` and `flatten(test[:, column_counter])`. Use those instead of performing these operations again.